stellargraph/data/explorer.py

```diff
@@ -789,3 +793,136 @@ def _check_neighbourhood_sizes(self, in_size, out_size):
             self._raise_error(
                 "The number of hops for the in and out neighbourhoods must be the same."
             )
+
+
+@experimental(reason="requires more thorough testing and documentation")
+class TemporalRandomWalk(GraphWalk):
+    """
+    Performs uniform temporal random walks on the given graph
+    """
+
+    def run(
+        self,
+        nodes=None,
+        n=None,
+        length=None,
+        bidirectional=False,
+        initial_edge_bias=None,
+        walk_bias=None,
+        seed=None,
+    ):
+        """
+        Perform a time respecting random walk starting from the root nodes.
+
+        Args:
+            nodes (list): The root nodes as a list of node IDs
+            n (int): Total number of random walks per root node
+            length (int): Maximum length of each random walk
+            bidirectional (bool, default False): Whether the walk extends on both direction.
+            initial_edge_bias (str, optional): distribution to use when choosing a random
+                initial temporal edge to start from. Available options are:
+                * None (default) - the initial edge is picked from a uniform distribution
+                * "exponential" - heavily biased towards more recent edges.
+            walk_bias (str, optional): distribution to use when choosing a random
+                neighbour to walk through. Available options are:
+                * None (default) - neighbours are picked from a uniform distribution
+                * "exponential" - exponentially decaying probability, resulting in a bias
+                    towards shorter time gaps
+            seed (int, optional): Random number generator seed; default is None
+
+        Returns:
+            List of lists of nodes ids for each of the random walks
+
+        """
+        self._check_common_parameters(nodes, n, length, seed)
+        np_rs = self._np_random_state if seed is None else np.random.RandomState(seed)
+
+        walks = []
+        for node in nodes:  # iterate over root nodes
```

Comment:
This could be a list comprehension:

```python
return [
    self._walk(node, length, bidirectional, initial_edge_bias, walk_bias, np_rs)
    for node in nodes
    for _ in range(n)
]
```